src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -83,18 +91,38 @@ public class ExtensionsRunner {
      */
     public ExtensionsRunner() throws IOException {}
 
+    /**
+     * Instantiates a new Extensions Runner.
+     *
+     * @throws IOException if the runner failed to read settings or API.
+     */
+    public ExtensionsRunner() throws IOException {}
+
     private ExtensionSettings readExtensionSettings() throws IOException {
         File file = new File(ExtensionSettings.EXTENSION_DESCRIPTOR);
         ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
-        ExtensionSettings extensionSettings = objectMapper.readValue(file, ExtensionSettings.class);
-        return extensionSettings;
+        return objectMapper.readValue(file, ExtensionSettings.class);
+    }
+
+    private ExtensionApi readExtensionApi() throws IOException {
+        File file = new File(ExtensionApi.EXTENSION_API_DESCRIPTOR);
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+        return objectMapper.readValue(file, ExtensionApi.class);
+    }
```

Comment:
Do you think its worth having its own class? This helps ExtensionsRunner to be simple and lean (and testable).

Reply:
Good question.  I agree that moving this (and corresponding tests) out of `ExtensionsRunner` is a good idea, but  I'm not sure it's worth a whole class just for this. There are some other options:
 - Move this to the `ExtensionApi` class
 - Create a new `YAMLReader` class and move both this and other similar methods into it

@owaiskazi19 any thoughts on this?

Reply:
Option 1 to move it to `ExtensionApi` class looks good to me here if we don't have any similar methods reading YAML file. (AFAIK this is the only one)

Reply:
On further review as I encountered this -- I actually think this is a temporary method given the eventual intent to get the API from a spec file.  So both this and the `readExtensionSettings()` are placeholders for better non-YAML functionality so I think they are fine to leave as-is.

Reply:
And on even further review, I moved this "temporary" code into `EstensionRestApi` class as a static method `readFromYaml()`, as well as moving the YML file to the class path in `src/main/resources` as an example to whoever does #86.

Reply:
Thanks Dan!